pytorch_lightning/loggers/neptune.py

```diff
@@ -114,160 +140,229 @@ def any_lightning_module_function_or_hook(self):
+            class LitModel(LightningModule):
+                def training_step(self, batch, batch_idx):
+                    # log metrics
+                    acc = ...
+                    self.logger.experiment['train/acc'].log(acc)
 
+                    # log images
+                    img = ...
+                    self.logger.experiment['train/misclassified_images'].log(File.as_image(img))
         """
+        return self.run
 
-        # Note that even though we initialize self._experiment in __init__,
-        # it may still end up being None after being pickled and un-pickled
-        if self._experiment is None:
-            self._experiment = self._create_or_get_experiment()
-
-        return self._experiment
+    @property
+    def run(self) -> Run:
+        if self._run_instance is None:
+            try:
+                self._run_instance = neptune.init(
+                    project=self._project,
+                    api_token=self._api_key,
+                    run=self._run_to_load,
+                    name=self._name,
+                    **self._neptune_run_kwargs,
+                )
+                self._run_instance['source_code/integrations/neptune-pytorch-lightning'] = __version__
+            except NeptuneLegacyProjectException as e:
+                raise TypeError(f"""
+                    Project {self._project} has not been imported to new structure yet.
+                    You can still integrate it with `NeptuneLegacyLogger`.
+                    """) from e
+
+        return self._run_instance
 
     @rank_zero_only
     def log_hyperparams(self, params: Union[Dict[str, Any], Namespace]) -> None:
+        r"""
+        Log hyper-parameters to the run.
+
+        Params will be logged using the ``param__`` scheme, for example: ``param__batch_size``, ``param__lr``.
+
+        **Note**
+
+        You can also log parameters by directly using the logger instance:
+        ``neptune_logger.experiment['model/hyper-parameters'] = params_dict``.
+
+        In this way you can keep hierarchical structure of the parameters.
+
+        Args:
+            params: `dict`.
+                Python dictionary structure with parameters.
+
+        Example::
+
+            from pytorch_lightning.loggers import NeptuneLogger
+
+            PARAMS = {
+                'batch_size': 64,
+                'lr': 0.07,
+                'decay_factor': 0.97
+            }
+
+            neptune_logger = NeptuneLogger(
+                api_key='ANONYMOUS',
+                close_after_fit=False,
+                project='common/new-pytorch-lightning-integration'
+            )
+
+            neptune_logger.log_hyperparams(PARAMS)
+        """
         params = self._convert_params(params)
         params = self._flatten_dict(params)
         for key, val in params.items():
-            self.experiment.set_property(f'param__{key}', val)
+            self.run[f'param__{key}'] = val
```

Comment:
@Herudaio do we want naming scheme `param__{key}` in the new logger? Why not `params/{key}`?

Reply:
Also, @shnela cover with tests